### DISCUSSION TITLE
How to configure ejabberd to use SQL Server with integrated authentication?

### DISCUSSION THREAD

**Discussion by petertirrell** [processone/ejabberd#3978] [Questions] [answered]
I currently am using ejabberd connected to a Microsoft SQL Server database. I am using a SQL Server login with username/password, per the documentation here: https://docs.ejabberd.im/admin/configuration/database/#microsoft-sql-server

How can I configure the ejabberd SQL Server storage to use SQL Server integrated security instead? I've not found documentation yet to address this.

A couple of ideas I have are to have the ejabberd service run as a kerberos/AD account, and configure the sql_server option as an ODBC connection string and use the integrated security option. Or to define the sql_username as the integrated login name, but remove the sql_password option.

Has anyone else tried doing this?

**Comment by nosnilmot** [comment, marked as answer]
This is almost certainly not currently possible (`sql_server` must be a hostname or IP address), but with the next ejabberd release (23.x for some value of x > 1) you will have more flexibility to specify SQL Server ODBC connection strings to refer to ODBC connections defined externally in system/user `odbcinst.ini` and `odbc.ini`

You can see a preview of the documentation for this at https://github.com/processone/docs.ejabberd.im/blob/61783e2d9ffc238d20026bd0c7027695bc00ff8f/content/admin/configuration/database.md#microsoft-sql-server

Note: I have not tested anything related to **Integrated Authentication** specifically, but if you can get that working in an ODBC configuration with `sqlcmd` / `isql` then I see no reason it would not work with ejabberd.
